pkg/utils/termination/suite_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package termination
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider/fake"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "knative.dev/pkg/logging/testing"
+
+	"sigs.k8s.io/karpenter/pkg/apis"
+	"sigs.k8s.io/karpenter/pkg/operator/scheme"
+	. "sigs.k8s.io/karpenter/pkg/test/expectations"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/test"
+)
+
+var (
+	ctx           context.Context
+	env           *test.Environment
+	cloudProvider *fake.CloudProvider
+)
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "NodeClaimUtils")
+}
+
+var _ = BeforeSuite(func() {
+	env = test.NewEnvironment(scheme.Scheme, test.WithCRDs(apis.CRDs...))
+	cloudProvider = fake.NewCloudProvider()
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = AfterEach(func() {
+	cloudProvider.Reset()
+	ExpectCleanedUp(ctx, env.Client)
+})
+
+var _ = Describe("NodeClaimUtils", func() {
+	var nodeClaim *v1beta1.NodeClaim
+	BeforeEach(func() {
+		nodeClaim = test.NodeClaim()
+		cloudProvider.CreatedNodeClaims[nodeClaim.Status.ProviderID] = nodeClaim
+	})
+	It("should update the status condition on NodeClaim as Terminating", func() {
+		nodeClaim.StatusConditions().SetFalse(v1beta1.ConditionTypeTerminating, "NotTerminating", "NotTerminating")
```

Comment:
nit: Do we even need to set the StatusCondition here to False? Doesn't seem like a realistic scenario here for what will actually happen in the controller; only because we are never setting this status condition to false. It's only not existing or set to True